src/PSRule/Pipeline/SourcePipeline.cs

```diff
@@ -352,10 +342,28 @@ private static SourceFile[] IncludeFile(string path, string helpPath)
             return new SourceFile[] { new SourceFile(path, null, GetSourceType(path), helpPath) };
         }
 
-        private static SourceFile[] IncludePath(string path, string helpPath, string moduleName)
+        private static SourceFile[] IncludePath(string path, string helpPath, string moduleName, bool excludeDefaultRulePath)
+        {
+            if (!excludeDefaultRulePath)
+            {
+                var allFiles = System.IO.Directory.EnumerateFiles(path, "*.Rule.*", SearchOption.AllDirectories);
+                return GetSourceFiles(allFiles, helpPath, moduleName);
+            }
+
+            var filteredFiles = FilterFiles(
+                path,
+                directoryFilter: dir => !dir.Contains(
+                    DefaultRulePath.TrimEnd(Path.AltDirectorySeparatorChar),
+                    StringComparison.OrdinalIgnoreCase),
+                filePattern: "*.Rule.*");
+
+            return GetSourceFiles(filteredFiles, helpPath, moduleName);
+        }
```

Comment:
Used this workaround to filter out `.ps-rule/` to fix the bug where files are searched twice. The excludeDefaultRulePath param basically ensures that directory is not searched, since by default `.ps-rule/` is searched during `Build()`.

Reply:
Let me know if we need to change this behaviour. The other option I considered was just filtering directly from `EnumerateFiles`, but tried to create a solution where we can just ignore the directory on the initial scan.

Reply:
This part of the code is fairly tuned in terms of performance ops/ mem. I'll have a look.